lambda/simplif.ml

```diff
@@ -329,6 +329,25 @@ let simplify_exits lam =
    Assumes |args| = |params|.
 *)
 
+let exact_application {kind; params; _} args =
+  match kind with
```

Comment:
Cosmetic suggestion: this could be written a bit more compacty, matching on `kind` and `args` at the same time, using an or-pattern for `Curried, args | Tupled, [Lprim(Pmakeblock _, args, _)]`; and when guards for the conditions on the number of arguments (=> a single final case returning `None`).

Reply:
I kept the current presentation, which I think makes it easier for humans to reason about all the different cases. In particular, it is clear that we are exhaustive on `cases` but fragile on `args`, and the addition of the `Const_block` case was a simple extension of the existing structure of the code -- whereas with your proposal it does not fit nicely, as `args` has a different type there.